python/pfnet/cpfnet.pyx

```diff
@@ -1232,6 +1318,258 @@ cdef class Branch:
+            else:
+                return np.array(r)
+
+    property Q_km_series:
+        """ Reactive power flow at bus "k" towards bus "m" over the series impedance of the line (from -> to) (p.u.) (float or array). """
+        def __get__(self):
+            r = [cbranch.BRANCH_get_Q_km_series(self._c_ptr,NULL,t) for t in range(self.num_periods)]
+            if self.num_periods == 1:
+                return AttributeFloat(r[0])
+            else:
+                return np.array(r)
+
+    property P_mk_series:
+        """ Real power flow at bus "m" towards bus "k" over the series impedance of the line (to -> from) (p.u.) (float or array). """
+        def __get__(self):
+            r = [cbranch.BRANCH_get_P_mk_series(self._c_ptr,NULL,t) for t in range(self.num_periods)]
+            if self.num_periods == 1:
+                return AttributeFloat(r[0])
+            else:
+                return np.array(r)
+
+    property Q_mk_series:
+        """ Reactive power flow at bus "m" towards bus "k" over the series impedance of the line (to -> from) (p.u.) (float or array). """
+        def __get__(self):
+            r = [cbranch.BRANCH_get_Q_mk_series(self._c_ptr,NULL,t) for t in range(self.num_periods)]
+            if self.num_periods == 1:
+                return AttributeFloat(r[0])
+            else:
+                return np.array(r)
+
+    property P_k_shunt:
+        """ Real power flow into the shunt element at bus "k" (aka "from") (p.u.) (float or array). """
+        def __get__(self):
+            r = [cbranch.BRANCH_get_P_k_shunt(self._c_ptr,NULL,t) for t in range(self.num_periods)]
+            if self.num_periods == 1:
+                return AttributeFloat(r[0])
+            else:
+                return np.array(r)
+
+    property Q_k_shunt:
+        """ Reactive power flow into the shunt element bus "k" (aka "from") (p.u.) (float or array). """
+        def __get__(self):
+            r = [cbranch.BRANCH_get_Q_k_shunt(self._c_ptr,NULL,t) for t in range(self.num_periods)]
+            if self.num_periods == 1:
+                return AttributeFloat(r[0])
+            else:
+                return np.array(r)
+
+    property P_m_shunt:
+        """ Real power flow into the shunt element at bus "m" (aka "to") (p.u.) (float or array). """
+        def __get__(self):
+            r = [cbranch.BRANCH_get_P_m_shunt(self._c_ptr,NULL,t) for t in range(self.num_periods)]
+            if self.num_periods == 1:
+                return AttributeFloat(r[0])
+            else:
+                return np.array(r)
+
+    property Q_m_shunt:
+        """ Reactive power flow into the shunt element at bus "m" (aka "to") (p.u.) (float or array). """
+        def __get__(self):
+            r = [cbranch.BRANCH_get_Q_m_shunt(self._c_ptr,NULL,t) for t in range(self.num_periods)]
+            if self.num_periods == 1:
+                return AttributeFloat(r[0])
+            else:
+                return np.array(r)
+
+    property P_from_to:
```

Comment:
See point 6) There is excessive redundancy here. We need to say that this is an alias for something else, and there is no need to have code that uses deprecated C routines.